lecture_1/task_1/main.go

```diff
@@ -0,0 +1,88 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/pkg/errors"
+	"github.com/sethgrid/pester"
+	"io/ioutil"
+	"log"
+	"os"
+	"time"
+)
+
+type serverData struct {
+	Name string
+	Age int
+	Passed bool
+	Skills []string
+}
+
+const applicationsURL = "https://run.mocky.io/v3/f7ceece5-47ee-4955-b974-438982267dc8"
+
+func linearBackoff(retry int) time.Duration {
+	return time.Duration(retry) * time.Second
+}
+
+func containsJavaOrGo(skills []string) bool {
+	for _, val := range skills {
+		if val == "Java" || val == "Go" {
+			return true
+		}
+	}
+	return false
+}
+
+func writePeopleInFile(f *os.File, data []serverData) {
+	for _, val := range data {
+		if val.Passed && containsJavaOrGo(val.Skills){
+			var allSkills string
+			for idx, value := range val.Skills {
+				if idx != 0 {
+					allSkills += ", " + value
+				} else {
+					allSkills += " - " + value
+				}
+			}
+			defer f.WriteString(fmt.Sprint(val.Name + allSkills) + "\n")
```

Comment:
Here, defer is unnecessary, Vedran used it only as an example in a for loop.

Also, you can replace this code by:
`f.WriteString(fmt.Sprintf("%s - %s\n", userApplication.Name, <something>))`

Where `<something>` is GoLang standard library function for creating single `string` from `[]string`.
(Hint: take a look at GoLang `strings` standard library package)